src/foam/nanos/auth/LastModifiedAwareDAO.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
```

Comment:
LastModifiedAware and EnabledAware aren't but LastModifiedByAware is, and it seemed like they should be in the same directory. I thought about putting them in foam.pattern, but you generally think of patterns as being more complex. Perhaps just putting them in foam.dao would be better, and then add a constant to LastModifiedByAware for the PRINCIPAL_KEY, so that it's not directly tied to the auth system.